### PR TITLE
Added a getEntry() to BanList

### PR DESCRIPTION
**Pull request by zKoz210**
## Introduction
Adds direct access to BanEntry

### API changes
Added `BanList::getEntry(string $name): ?BanEntry`

## Backward compatibility
This PR should preserve 100% backward-compatibility

### CODE REVIEW

File: src/pocketmine/permission/BanList.php [hunk 1]
```diff
@@ -58,6 +58,23 @@ public function setEnabled(bool $flag){
 		$this->enabled = $flag;
 	}
 
+	/**
+	 * @param string $name
+	 *
+	 * @return BanEntry|null
+	 */
+	public function getEntry(string $name): ?BanEntry{
```
Comment (dktapps) [inline src/pocketmine/permission/BanList.php]: formatting issue

File: src/pocketmine/permission/BanList.php [hunk 2]
```diff
@@ -58,6 +58,23 @@ public function setEnabled(bool $flag){
...
+		$name = strtolower($name);
+		
+		$this->removeExpired();
+
+		if (!isset($this->list[$name])){
+			return null;
+		}
+
+		return $this->list[$name];
```
Comment (dktapps) [inline src/pocketmine/permission/BanList.php]: the `isset()` and this statement could be merged into a single statement using a null coalesce (`$this->list[$name] ?? null`)

**Review by dktapps — CHANGES_REQUESTED** [review]
Looks like a reasonable addition, but please see comments on the diff.

File: src/pocketmine/permission/BanList.php [hunk 3]
```diff
@@ -58,6 +58,19 @@ public function setEnabled(bool $flag){
...
+	 * @param string $name
+	 *
+	 * @return BanEntry|null
+	 */
+	public function getEntry(string $name) : ?BanEntry{
+		$name = strtolower($name);
+		
+		$this->removeExpired();
+
+		return $this->list[$name] ?? null;
```
Comment (dktapps) [inline src/pocketmine/permission/BanList.php]: the `strtolower()` can now be moved to here, and remove the first statement, since it's only used once.

**Review by SOF3 — APPROVED** [review]

**Review by dktapps — APPROVED** [review]

**Comment by dktapps** [comment]
I'd better check the EOF newline before I merge this - not really clear why it still shows up on the diff.

**Pull request merged**